### PR TITLE
Instagram verification

### PR DESCRIPTION
**Pull request by jowlee**
### Trello Card Link
na

### Description
NOTE: Can review code now, but would like to test after staging's BE identity service is put up before final review. 
This can be reviewed one commit at a time.

Adds instagram verification - [Spec](https://www.figma.com/file/5fHG3aUYmqaiyHzUJpXKeo/Misc-Features?node-id=499%3A8619)
* On signup adds an option for instagram verification to autopopulate
* On the desktop settings page, changes the twitter verification button into a button that pops up a modal. The modal allows the user to verify their account with instagram or twitter.
* On mobile on the account settings page, adds a subpage for verification w/ twitter or instagram. 
* The instagram button in signup and settings is up behind a remote config feature flag.

**NOTE**: I need to add the plus icon that's missing. 

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?
Could potentially affect login flow if the user is stuck in the instagram/twitter loading phase

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

This was tested by using the staging instagram credentials and running on local https://localhost:3001 - note that it needs to be https. It was run against a local identity server w/ the staging instagram credentials to retrieve the user's username and fetch their profile to populate signup and add the entry to the backend table. 
The edit profile twitter/instagram handles were tested by manually passing in instagramVerified and twitterVerified. 

NOTE: I would like to put this up on a subdomain to test w/ staging's identity service after the backend is approved.

### CODE REVIEW

File: src/components/general/InstagramButton.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  className?: string
+  textClassName?: string
+  iconClassName?: string
+  onClick?: () => void
+  onSuccess: (uuid: string, profile: any) => void
+  onFailure: (error: any) => void
+  style?: object
+  disabled?: boolean
+  text?: string
+}
```
Comment (raymondjacobson) [inline src/components/general/InstagramButton.tsx:20]: nit \n

File: src/components/general/InstagramButton.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  text
+}: InstagramAuthButtonProps) => {
+  return (
+    <InstagramAuth
+      onClick={onClick}
+      disabled={disabled}
+      onFailure={onFailure}
+      onSuccess={onSuccess}
+      profileUrl={`${IDENTITY_SERVICE}/instagram`}
+      // profileUrl={`http://localhost:7000/instagram`}
```
Comment (raymondjacobson) [inline src/components/general/InstagramButton.tsx:38]: remove?

File: src/containers/settings-page/components/TwitterAccountVerified.module.css [hunk 3]
```diff
@@ -1,10 +1,22 @@
...
+  height: 64px !important;
+  min-width: 273px !important;
 }
 
 .twitterTextContainer {
   text-align: left;
   padding: 2px 14px;
 }
+
+.twitterButton {
```
Comment (raymondjacobson) [inline src/containers/settings-page/components/TwitterAccountVerified.module.css:11]: ;o

File: src/containers/settings-page/components/desktop/VerificationModal.module.css [hunk 4]
```diff
@@ -0,0 +1,143 @@
+.wrapperClassName {
+  max-width: 736px;
+  width: 100%;
+  margin: auto;
+}
+
+.container {
+  border-top: 1px solid #F2F2F4;
```
Comment (raymondjacobson) [inline src/containers/settings-page/components/desktop/VerificationModal.module.css:8]: neutral light 8

File: src/containers/settings-page/components/desktop/VerificationModal.tsx [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+const VerificationModal = (props: VerificationModalProps) => {
+  const { handle, onInstagramLogin, onTwitterLogin } = props
+  const [error, setError] = useState('')
+  const [status, setStatus] = useState('')
+  const onClick = useCallback(() => setStatus(Status.LOADING), [setStatus])
+  const onFailure = useCallback(() => {
+    setError(messages.failure)
+    setStatus(Status.ERROR)
+  }, [setError, setStatus])
+  const instagramLogin = useCallback(
```
Comment (raymondjacobson) [inline src/containers/settings-page/components/desktop/VerificationModal.tsx:177]: \n

File: src/containers/settings-page/components/desktop/VerificationModal.tsx [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+    },
+    [handle, onTwitterLogin, setError]
+  )
+
+  const [isOpen, setIsOpen] = useState(false)
+  const onOpen = useCallback(() => setIsOpen(true), [setIsOpen])
+  const onClose = useCallback(() => {
+    setIsOpen(false)
+    setError('')
+  }, [setIsOpen, setError])
```
Comment (raymondjacobson) [inline src/containers/settings-page/components/desktop/VerificationModal.tsx:214]: \n

File: src/containers/sign-on/components/desktop/TwitterOverlay.js [hunk 7]
```diff
@@ -2,26 +2,42 @@ import React from 'react'
...
+      <div>
+        {'Verification'} <IconVerified className={styles.verified} />
+      </div>
+      <div className={styles.ifApplicable}>{'(if applicable)'}</div>
+    </div>
   ],
   manual: 'I’d rather fill out my profile manually'
 }
 
 const TwitterOverlay = props => {
```
Comment (raymondjacobson) [inline src/containers/sign-on/components/desktop/TwitterOverlay.js:36]: maybe worth renaming TwitterOverlay

File: src/containers/sign-on/components/mobile/ProfilePage.tsx [hunk 8]
```diff
@@ -108,6 +120,30 @@ const ProfilePage = (props: ProfilePageProps) => {
...
+          type: 'image/jpeg'
+        })
+        const file = await resizeImage(artworkFile)
+        const url = URL.createObjectURL(file)
+        setInstagramProfile(uuid, profile, { url, file })
+      } else {
+        setInstagramProfile(uuid, profile)
+      }
+    } catch (err) {
+      // Continue if error
```
Comment (raymondjacobson) [inline src/containers/sign-on/components/mobile/ProfilePage.tsx:142]: maybe log it out?

**Review by raymondjacobson — APPROVED** [review]
huge lift. this looks great. easy to read thru

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/jowlee-instagram-verification

**Pull request merged**